### PR TITLE
Add kubevirt-ipi workflow

### PR DESCRIPTION
**Pull request by bardielle**
Signed-off-by: Roy Golan <rgolan@redhat.com>

### CODE REVIEW

File: ci-operator/step-registry/ipi/conf/kubevirt/OWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+approvers:
+  - bardielle
+  - ravidbro
+  - nirarg
+  - rgolang
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/conf/kubevirt/OWNERS:5]: rgolang :+1:

Comment (rgolangh) [reply ci-operator/step-registry/ipi/conf/kubevirt/OWNERS:5]: :smile:

File: ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-commands.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+
+KUBEVIRT_BASE_DOMAIN="tenant1.origin-on-kubevirt.dev.openshift.com"
+KUBEVIRT_API_VIP=192.168.123.15
+KUBEVIRT_INGRESS_VIP=192.168.123.20
+KUBEVIRT_KUBECONFIG=${CLUSTER_DIR}/.kubevirt/kubeconfig
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-commands.sh:13]: this one should probably be $HOME/.kube/config

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 3]
```diff
@@ -21,6 +21,7 @@ case "${CLUSTER_TYPE}" in
 aws) export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred;;
 azure4) export AZURE_AUTH_LOCATION=${CLUSTER_PROFILE_DIR}/osServicePrincipal.json;;
 gcp) export GOOGLE_CLOUD_KEYFILE_JSON=${CLUSTER_PROFILE_DIR}/gce.json;;
+kubevirt) export KUBEVIRT_KUBECONFIG=${CLUSTER_PROFILE_DIR}/.kubevirt/kubeconfig;;
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:24]: I think we should change this to KUBECONFIG=${HOME}/.kube/config

File: ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-ref.metadata.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+	"owners": {
+		"approvers": [
+			"bardielle",
+			"ravidbro",
+			"nirarg",
+			"rgolang",
+			"chenyosef"
+		]
+	}
+}
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-ref.metadata.json:12]: missing new line

File: ci-operator/step-registry/ipi/kubevirt/OWNERS [hunk 5]
```diff
@@ -0,0 +1,6 @@
+approvers:
+  - bardielle
+  - ravidbro
+  - nirarg
+  - rgolang
+  - chenyosef
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/OWNERS:6]: missing new line + rgolang

File: ci-operator/step-registry/ipi/kubevirt/post/OWNERS [hunk 6]
```diff
@@ -0,0 +1,6 @@
+approvers:
+  - bardielle
+  - ravidbro
+  - nirarg
+  - rgolang
+  - chenyosef
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/post/OWNERS:6]: missing new line + rgolang

File: ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.metadata.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+	"owners": {
+		"approvers": [
+			"bardielle",
+			"ravidbro",
+			"nirarg",
+			"rgolang",
+			"chenyosef"
+		]
+	}
+}
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.metadata.json:12]: missing new line + rgolang

File: ci-operator/step-registry/ipi/kubevirt/pre/stableinitial/ipi-kubevirt-pre-stableinitial-chain.metadata.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+	"owners": {
+		"approvers": [
+			"bardielle",
+			"ravidbro",
+			"nirarg",
+			"rgolang",
+			"chenyosef"
+		]
+	}
+}
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/pre/stableinitial/ipi-kubevirt-pre-stableinitial-chain.metadata.json:12]: missing new line + rgolang

File: ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.yaml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+chain:
+  as: ipi-kubevirt-post
+  steps:
+  - chain: ipi-conf-kubevirt
+  - chain: ipi-deprovision
+  documentation: |-
+    The IPI cleanup step contains all steps that deprovision an OpenShift
+    cluster on GCP, provisioned by the `ipi-gcp-pre` chain.
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.yaml:8]: s/GCP/KubeVirt   and s/ipi-gcp-pre/ipi-kubevirt-pre

File: ci-operator/step-registry/ipi/kubevirt/pre/ipi-kubevirt-pre-chain.metadata.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+	"owners": {
+		"approvers": [
+			"bardielle",
+			"ravidbro",
+			"nirarg",
+			"rgolang",
+			"chenyosef"
+		]
+	}
+}
```
Comment (rgolangh) [inline ci-operator/step-registry/ipi/kubevirt/pre/ipi-kubevirt-pre-chain.metadata.json:12]: missing new line + rgolang

File: ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-chain.metadata.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+	"owners": {
+		"approvers": [
+			"bardielle",
+			"ravidbro",
+			"nirarg",
+			"rgolang",
+			"chenyosef"
+		]
+	}
+}
```
[the remark this answers is not in the document]
Comment (bardielle) [reply ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-chain.metadata.json:12]: 8rgolnag is in line 8

File: ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-ref.metadata.json (hunk 4 above)
Comment (bardielle) [reply ci-operator/step-registry/ipi/conf/kubevirt/ipi-conf-kubevirt-ref.metadata.json:12]: rgolang is in line 8

File: ci-operator/step-registry/ipi/kubevirt/OWNERS (hunk 5 above)
Comment (bardielle) [reply ci-operator/step-registry/ipi/kubevirt/OWNERS:6]: rgolang is in line 5

File: ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.metadata.json (hunk 7 above)
Comment (bardielle) [reply ci-operator/step-registry/ipi/kubevirt/post/ipi-kubevirt-post-chain.metadata.json:12]: rgolang is in line 8

File: ci-operator/step-registry/ipi/kubevirt/pre/ipi-kubevirt-pre-chain.metadata.json (hunk 10 above)
Comment (bardielle) [reply ci-operator/step-registry/ipi/kubevirt/pre/ipi-kubevirt-pre-chain.metadata.json:12]: rgolang is in line 8

File: ci-operator/step-registry/ipi/kubevirt/pre/stableinitial/ipi-kubevirt-pre-stableinitial-chain.metadata.json (hunk 8 above)
Comment (bardielle) [reply ci-operator/step-registry/ipi/kubevirt/pre/stableinitial/ipi-kubevirt-pre-stableinitial-chain.metadata.json:12]: rgolang is in line 8

**Pull request merged**